.github/workflows/deploy-standalone-plugins.yml

```diff
@@ -1,99 +1,154 @@
-name: Deploy standalone plugins to WordPress.org
+---
+name: Deploy Standalone Plugin(s) to WordPress.org
 
 on:
   release:
-    types: [published]
+    types:
+      - published
   workflow_dispatch:
     inputs:
-      slug:
+      plugin:
         type: string
         description: 'The slug of the plugin to deploy'
+        required: true
       dry-run:
         type: boolean
         description: 'Debug mode (run without publishing).'
         default: false
+
+permissions: {}
+
 jobs:
-  release:
-    name: Prepare Deployment
+  pre-run:
+    name: Pre-run
     runs-on: ubuntu-latest
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      matrix: ${{ steps.set-matrix.outputs.plugins }}
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
-      - name: Setup Node.js (.nvmrc)
-        uses: actions/setup-node@v3
-        with:
-          node-version-file: '.nvmrc'
-          cache: npm
-      - name: Install npm dependencies
-        run: npm ci
```

Comment:
You're moving these steps to `deploy` further below, which runs for every plugin to deploy. Maybe that's the only way, but would it be possible to instead leave these actions here?

The reason is that checking out the repository and installing dependencies will be the same for every plugin. So it would be more efficient to do it once in the beginning, regardless of how many or which plugins are being deployed afterwards.

Reply:
@felixarntz These steps are here just to install NPM deps so that we can run a few CLI commands to get the plugin version and path to the matrix.

In the previous setup, we were running `npm ci` in two places:
- release job
- deployment job(which is also a matrix)

In the current setup, we were running `npm ci`  only once:
- deploy job (which is a matrix)